gdb/rust-lang.c

```diff
@@ -289,7 +403,66 @@ rust_val_print (struct type *type, const gdb_byte *valaddr, int embedded_offset,
 	  goto generic_print;
       }
       break;
+    case TYPE_CODE_UNION:
+    {
+      int j, nfields, first_field, is_tuple;
+      struct type *variant_type;
+      struct disr_info disr;
+      struct value_print_options opts;
+
+      opts = *options;
+      opts.deref_ref = 0;
+
+      disr = rust_get_disr_info(type, valaddr, embedded_offset, address, val);
+
+      first_field = 1;
+      variant_type = TYPE_FIELD_TYPE (type, disr.field_no);
+      nfields = TYPE_NFIELDS (variant_type);
+
+      is_tuple = rust_tuple_variant_type_p (variant_type);
+
+      if (nfields > 1) {
+        /* In case of a non-nullary variant, we output `Foo(x,y,z)`. */
+        if (is_tuple) {
+          fprintf_filtered (stream, "%s(", disr.name);
+        } else {
+          /* struct variant */
+          fprintf_filtered (stream, "%s{", disr.name);
+        }
+
+      } else {
+        /* In case of a nullary variant like `None`, just output the name. */
+        fprintf_filtered (stream, "%s", disr.name);
+        break;
+      }
+
+      for (j = 1; j < TYPE_NFIELDS (variant_type); j++) {
+
+        if (!first_field) {
+          fputs_filtered (", ", stream);
+        }
```

Comment:
In fact, `rustfmt` doesn't even attempt to reflow large variants when it finds it. That's a bug, of course, but the fact that it wasn't even considered indicates how rare this is in Rust.